src/coreclr/vm/nativememorynative.cpp

```diff
@@ -0,0 +1,106 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+// File: NativeMemoryNative.cpp
+//
+
+#include "config.h"
+#include "common.h"
+#include "nativememorynative.h"
+
+/* static */ void* QCALLTYPE NativeMemoryNative::AlignedAlloc(size_t alignment, size_t size)
+{
+    QCALL_CONTRACT;
+
+    void* result = nullptr;
+
+    BEGIN_QCALL;
+
+#if HAVE_ALIGNED_ALLOC
+    result = aligned_alloc(alignment, size);
```

Comment:
Notably this is in C11 and C++17.

For C++17, it comes from `cstdlib` which we don't currently include (and which isn't trivial to include today).